### DISCUSSION TITLE
On_ready callback for setup_dap_main_class_configs()

### DISCUSSION THREAD

**Discussion by mikehaertl** [mfussenegger/nvim-jdtls#415] [Ideas]
I wonder if it would make sense to be able to pass a callback to `setup_dap_main_class_configs()`. It would be called when the configuration is ready.

I may do things wrong but right now to start debugging I do:

 * Call `setup_dap_main_class_configs()` or `:JdtRefreshDebugConfigs`
 * Wait ...
 * Call `require('dap').continue()` to start debugging

I can't put both into a function because `continue()` has to wait until the configuration is ready.

With a callback this could be streamlined to:

```lua
require('jdtls').setup_dap_main_class_configs({
  on_ready = function(config)
    require('dap').continue()
  end
})
```

**Comment by mfussenegger** [comment]
Makes sense to me. Opened up https://github.com/mfussenegger/nvim-jdtls/pull/419
